site/components/MdxComponents/MdxComponents.tsx

```diff
@@ -63,27 +63,45 @@ export const components = {
   ),
   h3: ({ id, ...props }) => {
     return (
-      <span id={id} style={{ scrollMarginTop: 100 }}>
-        <Text
-          as="h3"
-          marginBottom="2"
-          marginTop="11"
-          variant="title2"
-          weight="semibold"
-          {...props}
-        />
-      </span>
+      <Box
+        as="span"
+        display="block"
+        id={id}
+        marginBottom="2"
+        marginTop="11"
+        style={{ scrollMarginTop: 100 }}
+      >
+        <Link href={`#${id}`}>
+          <Text
+            as="h3"
+            display="inline"
+            variant="title2"
+            weight="semibold"
+            {...props}
+          />
+        </Link>
+      </Box>
     );
   },
-  h4: props => (
-    <Text
-      as="h4"
+  h4: ({ id, ...props }) => (
+    <Box
+      as="span"
```

Comment:
could we use `Box` as `div`, instead of `span`? That way we don't need to also add `display` support to the `Box`.

Reply:
Updated with responsive scroll top space keep 12 pixels away from the header.